test/Conversion/AMDGPU/load_store.mlir

```diff
@@ -1,16 +1,19 @@
 // RUN: triton-opt %s -split-input-file --convert-triton-gpu-to-llvm | FileCheck --check-prefixes=CHECK,GCN %s
 
-// Check load instructin doesn't generate incorrect bitcast.
+// Check load instruction doesn't generate incorrect bitcast.
 module attributes {"triton_gpu.num-warps" = 4 : i32} {
   // CHECK-LABEL: @test_float16_bitcast
   func public @test_float16_bitcast(%arg0: !tt.ptr<f16> {tt.divisibility = 16 : i32}, %arg1: !tt.ptr<f16> {tt.divisibility = 16 : i32}) {
     %true = arith.constant true
     %0 = tt.get_program_id {axis = 0 : i32} : i32
     %1 = tt.addptr %arg0, %0 : !tt.ptr<f16>, i32
     %2 = tt.load %1 {cache = 1 : i32, evict = 1 : i32, isVolatile = false} : f16
+    // GCN-NOT: llvm.bitcast {{.*}} : {{.*}}32{{.*}}16
+    // GCN-NOT: llvm.bitcast {{.*}} : {{.*}}16{{.*}}32
     // GCN: llvm.bitcast {{.*}} : i16 to f16
+    // GCN-NOT: llvm.bitcast {{.*}} : {{.*}}32{{.*}}16
+    // GCN-NOT: llvm.bitcast {{.*}} : {{.*}}16{{.*}}32
```

Comment:
I guess second time `GCN-NOT` is redundant. But perhaps check for `llvm.store` will be useful here.

Reply:
Maybe add check for load as well?

P.s. I thought about checks like this, but hesitate to add them because this test was originally dedicated to bitcast correctness

Reply:
Yes, the test is was added as a regression testing for the fixed bitcast creation. And since the bitcast was fixed in both load and store it seems to be a good idea to add both load and store and both their bitcasts.